### PR TITLE
Update and fix CI workflows

### PR DESCRIPTION
**Pull request by bmoffatt**
*Issue #, if available:*

*Description of changes:*

* added github action workflows for clang-format/clang-tidy/"the build", fixed any reccomentations
* updated the aws codebuild buildspecs and Dockerfiles to get their workflows running again
* added a sleep in integration tests to mitigate [Lambda Function States](https://aws.amazon.com/blogs/compute/coming-soon-expansion-of-aws-lambda-states-to-all-functions/) causing test failures

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/lambda-runtime/runtime.h [hunk 1]
```diff
@@ -163,8 +163,6 @@ class runtime {
         std::string const& url,
         std::string const& request_id,
         invocation_response const& handler_response);
-
-private:
```
Comment (marcomagdy) [inline include/aws/lambda-runtime/runtime.h:167]: What's the reason for making this public?

**Review by marcomagdy — COMMENTED** [review]
Thank you for taking care of this. 
It all looks good. I left one comment/question.

File: include/aws/lambda-runtime/runtime.h (hunk 1 above)
Comment (bmoffatt) [reply include/aws/lambda-runtime/runtime.h:167]: clang-tidy pointed out that these were already made private a few lines up

ref: https://github.com/bmoffatt/aws-lambda-cpp/runs/4305830378?check_suite_focus=true
```
/home/runner/work/aws-lambda-cpp/aws-lambda-cpp/include/aws/lambda-runtime/runtime.h:167:1: error: redundant access specifier has the same accessibility as the previous access specifier [readability-redundant-access-specifiers,-warnings-as-errors]
private:
^~~~~~~~
/home/runner/work/aws-lambda-cpp/aws-lambda-cpp/include/aws/lambda-runtime/runtime.h:159:1: note: previously declared here
private:
^
```

Comment (marcomagdy) [reply include/aws/lambda-runtime/runtime.h:167]: Ah! That's what I get for looking at this on a 5" phone screen.

**Review by marcomagdy — APPROVED** [review]

**Comment by marcomagdy** [comment]
Please squash before merging. Thanks again.

**Pull request merged**